WordPress/src/main/java/org/wordpress/android/ui/mysite/SiteItemsBuilder.kt

```diff
@@ -3,20 +3,69 @@ package org.wordpress.android.ui.mysite
 import org.wordpress.android.R
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.ui.mysite.MySiteItem.CategoryHeader
+import org.wordpress.android.ui.mysite.MySiteItem.ListItem
+import org.wordpress.android.ui.utils.ListItemInteraction
 import org.wordpress.android.ui.utils.UiString.UiStringRes
 import javax.inject.Inject
 
 class SiteItemsBuilder
 @Inject constructor(
-    private val siteCategoryItemBuilder: SiteCategoryItemBuilder
+    private val siteCategoryItemBuilder: SiteCategoryItemBuilder,
+    private val siteListItemBuilder: SiteListItemBuilder
 ) {
-    fun buildSiteItems(site: SiteModel): List<MySiteItem> {
+    fun buildSiteItems(site: SiteModel, onClick: ListItemInteraction): List<MySiteItem> {
         val siteItems = mutableListOf<MySiteItem>()
```

Comment:
Maybe we can make this whole code block a little bit more readable if we use `mutableListOf<MySiteItem>().apply` here instead of calling `siteItems` multiple times. What do you think?

Reply:
that's a great idea 👍

Reply:
Didn't know `listOfNotNull` was a thing, it seems perfect to this!